detect_secrets/core/baseline.py

```diff
@@ -299,6 +335,67 @@ def _get_git_tracked_files(rootdir='.'):
     return output
 
 
+def _get_git_diff_files(rootdir='.', from_commit=None):
+    """We may want to only return tracked files which have changed
+    since a certain commit ref.
+
+    :type rootdir: str
+    :param rootdir: root directory of where you want to list files from
+
+    :type from_commit: str
+    :param from_commit: reference of the commit to diff against
+
+    :rtype: set|None
+    :returns: filepaths to files which have changed since from_commit (locally)
+    """
+    output = []
+    try:
+        with open(os.devnull, 'w') as fnull:
+            git_files = subprocess.check_output(
+                [
+                    'git',
+                    '--no-pager',
+                    'diff',
+                    '--name-only',
+                    from_commit,
+                    rootdir,
+                ],
+                stderr=fnull,
+            )
+        for filename in git_files.decode('utf-8').split():
+            # As opposed to ls-files which gives path relative to -C, diff gives
+            # paths from git root directory
+            relative_path = util.get_relative_path_if_in_cwd('.', filename)
+            if relative_path:
+                output.append(relative_path)
+    except subprocess.CalledProcessError:
+        pass
```

Comment:
Nit: Might be better practice to only wrap the relevant `subprocess` call in a `try:` `except:`